Migrate Phases of Terraform Adoption WAF article (#635)
hashicorp/terraform-docs-common: website/docs/cloud-docs/recommended-practices/phases/index.mdx

---
page_title: Phases of Terraform Adoption
description: Evolve your Terraform strategy as adoption grows within your organization
---

# Phases of Terraform Adoption

> As more of your organization adopts Terraform, your infrastructure provisioning workflows must change and adapt. The workflows suitable for individual practitioners may not scale to larger enterprises. This guide will help you plan your organization's Terraform adoption strategy and presents workflow considerations that you should keep in mind to support future scale. This guide focuses on challenges faced by larger organizations, but we recommend implementing each practice as early as you can to help you scale smoothly. 

Reviewer: ```suggestion
As more of your organization adopts Terraform, your infrastructure provisioning workflows must change and adapt. The workflows suitable for individual practitioners may not scale to larger enterprises. This guide will help you plan your organization's Terraform adoption strategy and presents workflow considerations that you should keep in mind to support future scaling. This guide focuses on challenges faced by larger organizations, but we recommend implementing each practice as early as you can to help you scale smoothly. 
```

> As more of your organization adopts Terraform, your infrastructure provisioning workflows will need to change and adapt. The workflows that are suitable for individual practitioners may not scale to larger enterprises. This guide will help you plan your organization's Terraform adoption strategy and presents workflow considerations that you should keep in mind to support future scale. This guide focuses on challenges faced by larger organizations, but we recommend implementing each practice as early as you can to help you scale smoothly. 

Reviewer 2: ```suggestion
As more of your organization adopts Terraform, your infrastructure provisioning workflows must change and adapt. The workflows suitable for individual practitioners may not scale to larger enterprises. This guide will help you plan your organization's Terraform adoption strategy and presents workflow considerations that you should keep in mind to support future scale. This guide focuses on challenges faced by larger organizations, but we recommend implementing each practice as early as you can to help you scale smoothly. 
```

## Adopt

An individual practitioner can establish strong foundational practices that support future scale and make Terraform operations predictable and secure. 

[Learn how to adopt Terraform](/terraform/cloud-docs/recommended-practices/phases/adopt)

## Collaborate

> Multiple developers working on the same codebase introduces a new set of challenges, but solutions such as remote state backends help ease collaboration and coordinate execution.

Reviewer 2: ```suggestion
Multiple developers working on the same codebase introduce new challenges, but solutions such as remote state backends help ease collaboration and coordinate execution.
```

[Learn how to collaborate with Terraform](/terraform/cloud-docs/recommended-practices/phases/collaborate).

## Scale

As Terraform usage expands across your organization, you will need to decide how to define boundaries of infrastructure ownership.

> You will also need to decide on a cloud deployment strategy based on your organization's practices and needs. Possible approaches include using a single account in a single cloud provider, a hybrid or multi-cloud approach, or to divide up resources across accounts by environment.  Regardless of your implementation, Terraform lets you manage your infrastructure with a consistent workflow.

Reviewer 2: ```suggestion
You must also decide on a cloud deployment strategy based on your organization's practices and needs. Possible approaches include using a single account in a single cloud provider, a hybrid or multi-cloud approach, or dividing up resources across accounts by environment. Terraform lets you manage your infrastructure with a consistent workflow regardless of your implementation.
```

[Learn how to scale Terraform](/terraform/cloud-docs/recommended-practices/phases/scale).

## Govern

> As your teams grow, a common operational challenge is deciding how to enforce your organization's standards and practices. Using codified, automated policy enforcement with Sentinel or OPA ensures consistent application of your standards.

Reviewer: ```suggestion
As your teams grow, a common operational challenge is deciding how to enforce your organization's standards and practices. Using codified and automated policy enforcement with Sentinel or OPA ensures consistent application of your standards.
```
